react-frontend/src/reducers/loginReducer.js

```diff
@@ -0,0 +1,28 @@
+import actionGenerator from "utils/actionGenerator";
+
+const actionStatus = actionGenerator("LOGIN");
+
+const defaultState = {
+  status: null,
+  data: { token: null },
+  error: { message: null },
+};
+
+export default (state = defaultState, action) => {
+  switch (action.type) {
+    case actionStatus.success:
+      return {
+        ...state,
+        status: action.payload.status,
+        data: action.payload.value,
+      };
```

Comment:
Consider a scenario, first login attempt fails and then succeeds in second attempt. Do we need any changes here? Think about possible scenarios!